rules-reviewed/openshift/mail.windup.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0"?>
+<ruleset id="mail"
+    xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This is a ruleset for detecting Mail API usage when migrating an application to a cloud environment.
+        </description>
+        <dependencies>
+          <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final"/>
+          <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="java"/>
+        <sourceTechnology id="java-ee"/>
+        <targetTechnology id="cloud-readiness"/>
+        <tag>mail</tag>
+    </metadata>
+    <rules>
+        <rule id="mail-00000">
+            <when>
+                    <javaclass references="javax.mail.{*}">
+                        <location>CONSTRUCTOR_CALL</location>
+                        <location>IMPORT</location>
+                        <location>INHERITANCE</location>
+                        <location>VARIABLE_DECLARATION</location>
+                    </javaclass>
+            </when>
+            <perform>
+              <hint title="JavaMail: direct access of an SMTP server" effort="3" category-id="cloud-optional">
```

Comment:
It is true that the hint message supposes something it is not in the condition. If we are suggesting something about standard SMTP port could be blocked, it could search for that construct string IMO.
This rule should be more about potentialy identifying or reviewing SMTP or mail configurations.
Google/SAP and other cloud providers supply some sort of Java Mail services directly in its cloud platform.
Even Openshift supports that see https://blog.openshift.com/outbound-mail-ports-are-now-open-for-business-on-openshift/